### PR TITLE
telegraf: patch go.mod, remove runVend

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/monitoring/telegraf/gomod.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/go.mod b/go.mod
+index 1f6bd6b8..c2090ddb 100644
+--- a/go.mod
++++ b/go.mod
+@@ -110,7 +110,7 @@ require (
+ 	github.com/safchain/ethtool v0.0.0-20200218184317-f459e2d13664
+ 	github.com/samuel/go-zookeeper v0.0.0-20180130194729-c4fab1ac1bec // indirect
+ 	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b // indirect
+-	github.com/shirou/gopsutil v2.20.5+incompatible
++	github.com/shirou/gopsutil v2.20.6+incompatible
```
Comment (Mic92) [inline pkgs/servers/monitoring/telegraf/gomod.patch:10]: Did they solve the vendoring problem in this release? I can make a telegraf PR for this as well.

Comment (zowoq) [reply pkgs/servers/monitoring/telegraf/gomod.patch:10]: Yes, the vendoring was fixed in https://github.com/shirou/gopsutil/pull/889.

File: pkgs/servers/monitoring/telegraf/default.nix [hunk 2]
```diff
@@ -15,10 +15,11 @@ buildGoModule rec {
     sha256 = "045wjpq29dr0s48ns3a4p8pw1j0ssfcw6m91iim4pkrppj7bm2di";
   };
 
-  runVend = true;
-  vendorSha256 = "0c2sayg49b2rq3fnrbf741b6zy8byhwxlnxkhf5160gzqn6jy2rw";
+  patches = [
+    ./gomod.patch
```
Comment (Mic92) [inline pkgs/servers/monitoring/telegraf/default.nix:19]: Could you use fetchpatch here?

https://github.com/influxdata/telegraf/commit/a0d0c80c373dd1896d449e9f3534a157d78fc866.patch

This is the PR:

https://github.com/influxdata/telegraf/pull/7988

Comment (zowoq) [reply pkgs/servers/monitoring/telegraf/default.nix:19]:
```
vendor/github.com/shirou/gopsutil/host/host_bsd.go:25:13: undefined: unix.SysctlTimeval
``

Looks like it works with 2.20.6 and breaks with a different issue on 2.207.

Comment (Mic92) [reply pkgs/servers/monitoring/telegraf/default.nix:19]: Ok. I will have a look: https://github.com/shirou/gopsutil/issues/591

Comment (Mic92) [reply pkgs/servers/monitoring/telegraf/default.nix:19]: Bumped to 2.20.6: Let's see if that makes CI happy: https://github.com/influxdata/telegraf/commit/63e1f41d8ff246d191d008ff7f69d69cc34b4fae.patch

Comment (zowoq) [reply pkgs/servers/monitoring/telegraf/default.nix]: Builds with patch, generates the same `vendorSha256`.

Comment (Mic92) [reply pkgs/servers/monitoring/telegraf/default.nix]: Looks this patch breaks FreeBSD now.... https://app.circleci.com/pipelines/github/influxdata/telegraf/1327/workflows/6b7efed4-74fb-45bf-a3c7-2978f9bdf5bb/jobs/62496

However I think this should be good enough for nixpkgs.

**Pull request merged**